gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/api/AbstractConfigurablePluginManager.java

```diff
@@ -44,17 +44,32 @@
 
     @Override
     public String getSchema(String pluginId) throws IOException {
-        return getFirstFile(pluginId, SCHEMAS_DIRECTORY);
+        return getSchema(pluginId, false);
+    }
+
+    @Override
+    public String getSchema(String pluginId, boolean includeNotDeployed) throws IOException {
+        return getFirstFile(pluginId, SCHEMAS_DIRECTORY, includeNotDeployed);
     }
 
     @Override
     public String getSchema(String pluginId, String subFolder) throws IOException {
-        return getFirstFile(pluginId, String.format("%s/%s", SCHEMAS_DIRECTORY, subFolder));
+        return getSchema(pluginId, subFolder, false);
+    }
+
+    @Override
+    public String getSchema(String pluginId, String subFolder, boolean includeNotDeployed) throws IOException {
+        return getFirstFile(pluginId, String.format("%s/%s", SCHEMAS_DIRECTORY, subFolder), includeNotDeployed);
     }
 
     @Override
     public String getIcon(String pluginId) throws IOException {
-        T plugin = get(pluginId);
+        return getIcon(pluginId, false);
+    }
+
+    @Override
+    public String getIcon(String pluginId, boolean includeNotDeployed) throws IOException {
+        T plugin = get(pluginId, includeNotDeployed);
         Map<String, String> properties = plugin.manifest().properties();
```

Comment:
Shouldn't we handle the case where no plugin is found ?

Reply:
Yep was not done before but make sense